### PR TITLE
Add unit tests for Fused NovoGrad

### PR DESCRIPTION
**Pull request by sudhakarsingh27**
1. Added unit tests for fused-novograd
2. Fixed issue that noop flag tensor was not on the same device as other tensors.
3. Fixed issue that even with all tensors on the same device, cuda stream is on a different device than tensors.

### CODE REVIEW

**Comment by slayton58** [comment]
\cc @ptrblck

File: tests/L0/run_optimizers/test_fused_novograd.py [hunk 1]
```diff
@@ -0,0 +1,168 @@
+import torch
+from torch.optim import Optimizer
+import math
+import apex
+import unittest
+
+from .test_fused_optimizer import TestFusedOptimizer
```
Comment (ptrblck) [inline tests/L0/run_optimizers/test_fused_novograd.py:7]: This import will fail with:
```python
ImportError: attempted relative import with no known parent package
```
Could you remove the dot at the beginning?

File: tests/L0/run_optimizers/test_fused_novograd.py [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+            tensors, self.options, self.tst_options
+        )
+
+        for _ in range(self.iters):
+            self.gen_grad(ref_param, tst_param)
+            ref_optim.step()
+            tst_optim.step()
+            max_abs_diff, max_rel_diff = self.get_max_diff(ref_param, tst_param)
+            self.assertLessEqual(max_abs_diff, self.max_abs_diff)
+            self.assertLessEqual(max_rel_diff, self.max_rel_diff)
```
Comment (ptrblck) [inline tests/L0/run_optimizers/test_fused_novograd.py:168]: The unittest execution seems to be missing via:
```python
if __name__ == '__main__':
    unittest.main()
```
Currently no tests will be run py `python test_fused_novograd.py -v`.

File: tests/L0/run_optimizers/test_fused_novograd.py [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+
+    def test_half(self):
+        self.gen_single_type_test(param_type=torch.float16)
+
+    @unittest.skipIf(torch.cuda.device_count()<2, "more than 1 GPU required")
+    def test_multi_device(self):
+        devices = ("cuda:1", "cuda:0")
+        for current_dev, tensor_dev in product(devices, devices):
+            with torch.cuda.device(current_dev):
+                print("Some info from the test", current_dev, tensor_dev)
```
Comment (ptrblck) [inline tests/L0/run_optimizers/test_fused_novograd.py:147]: Could you remove this `print` statement or is this needed for the test?

**Review by ptrblck — CHANGES_REQUESTED** [review]
Thanks for the PR! Some changes would be needed before we can merge it. :)

File: tests/L0/run_optimizers/test_fused_novograd.py (hunk 1 above)
Comment (sudhakarsingh27) [reply tests/L0/run_optimizers/test_fused_novograd.py]: Done.

File: tests/L0/run_optimizers/test_fused_novograd.py (hunk 2 above)
Comment (sudhakarsingh27) [reply tests/L0/run_optimizers/test_fused_novograd.py:167]: Right, added it now.

File: tests/L0/run_optimizers/test_fused_novograd.py (hunk 3 above)
Comment (sudhakarsingh27) [reply tests/L0/run_optimizers/test_fused_novograd.py]: I've removed it.

**Pull request merged**